includes/data-stores/class-wc-admin-reports-downloads-data-store.php

```diff
@@ -0,0 +1,375 @@
+			$where_filters[] = " {$lookup_table}.permission_id IN (
+			SELECT
+				DISTINCT {$wpdb->prefix}woocommerce_downloadable_product_permissions.permission_id
+			FROM
+				{$wpdb->prefix}woocommerce_downloadable_product_permissions
+			WHERE
+				{$wpdb->prefix}woocommerce_downloadable_product_permissions.order_id IN ({$included_orders})
+			)";
+		}
+
+		if ( $excluded_orders ) {
+			$where_filters[] = " {$lookup_table}.permission_id NOT IN (
+			SELECT
+				DISTINCT {$wpdb->prefix}woocommerce_downloadable_product_permissions.permission_id
+			FROM
+				{$wpdb->prefix}woocommerce_downloadable_product_permissions
+			WHERE
+				{$wpdb->prefix}woocommerce_downloadable_product_permissions.order_id IN ({$excluded_orders})
+			)";
+		}
+
+		$included_users = $this->get_included_users( $query_args );
+		$excluded_users = $this->get_excluded_users( $query_args );
+		if ( $included_users ) {
+			$where_filters[] = " {$lookup_table}.permission_id IN (
+			SELECT
+				DISTINCT {$wpdb->prefix}woocommerce_downloadable_product_permissions.permission_id
+			FROM
+				{$wpdb->prefix}woocommerce_downloadable_product_permissions
+			WHERE
+				{$wpdb->prefix}woocommerce_downloadable_product_permissions.user_id IN ({$included_users})
+			)";
+		}
+
+		if ( $excluded_users ) {
+			$where_filters[] = " {$lookup_table}.permission_id NOT IN (
+			SELECT
+				DISTINCT {$wpdb->prefix}woocommerce_downloadable_product_permissions.permission_id
+			FROM
+				{$wpdb->prefix}woocommerce_downloadable_product_permissions
+			WHERE
+				{$wpdb->prefix}woocommerce_downloadable_product_permissions.user_id IN ({$excluded_users})
+			)";
+		}
+
+		$included_ip_addresses = $this->get_included_ip_addresses( $query_args );
+		$excluded_ip_addresses = $this->get_excluded_ip_addresses( $query_args );
+		if ( $included_ip_addresses ) {
+			$where_filters[] = " {$lookup_table}.user_ip_address IN ('{$included_ip_addresses}')";
+		}
+
+		if ( $excluded_ip_addresses ) {
+			$where_filters[] = " {$lookup_table}.user_ip_address NOT IN ('{$excluded_ip_addresses}')";
+		}
+
+		$where_subclause = implode( " $operator ", $where_filters );
+		if ( $where_subclause ) {
+			$sql_query_params['where_clause']    .= " AND ( $where_subclause )";
+		}
+
+		$sql_query_params['from_clause'] .= " JOIN {$wpdb->prefix}woocommerce_downloadable_product_permissions as product_permissions ON {$lookup_table}.permission_id = product_permissions.permission_id";
+
+		return $sql_query_params;
+	}
+
+	/**
+	 * Returns comma separated ids of included ip address, based on query arguments from the user.
+	 *
+	 * @param array $query_args Parameters supplied by the user.
+	 * @return string
+	 */
+	protected function get_included_ip_addresses( $query_args ) {
+		$included_ips_str = '';
+
+		if ( isset( $query_args['ip_address_includes'] ) && is_array( $query_args['ip_address_includes'] ) && count( $query_args['ip_address_includes'] ) > 0 ) {
+			$included_ips_str = implode( "','", $query_args['ip_address_includes'] );
```

Comment:
Do we need to sanitize the input here at all or are we assuming that is all handled in `rest_validate_request_arg`?

Reply:
I've added escaping here to make sure, since that just validates the arg but does not sanitation.